### ISSUE TITLE
Automatic takeover of the pagemanager height from the CMS settings for the editor

### ISSUE THREAD

**Issue by frase-git** [cmb69/pagemanager_xh#74]
In `... \pagemanager\css\stylesheet.css` the height of the pagemanager is hardcoded (400px).
Of course you can easily adjust it. There is also a hint there, okay.

But I think it would be better if this height would be taken from the configuration of the editor height `$cf['editor']['height']`.
Who changes something there, has surely a good reason to do so. The page manager would also be adjusted (at least similarly).

To do this, it is probably enough to inject a single CSS rule into the header - right?

**Comment by cmb69** [comment]
Yeah, might be reasonable to use `$cf['editor']['height']`. And indeed, that could be solved by adding the rule to `$hjs`. That would still allow users to override the height in `stylesheet.css` if they desire so.

**Comment by cmb69** [comment]
Der Patch ist eigentlich trivial (hier mal mit hart-kodierter Höhe):
````diff
 classes/MainAdminController.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/classes/MainAdminController.php b/classes/MainAdminController.php
index 4659197..8389067 100644
--- a/classes/MainAdminController.php
+++ b/classes/MainAdminController.php
@@ -96,6 +96,7 @@ class MainAdminController
         }
         $hjs .= '<link rel="stylesheet" type="text/css" href="'
             . "{$this->pluginFolder}jstree/themes/{$this->config['treeview_theme']}/style.min.css" . '">';
+        $hjs .= '<style>#pagemanager {height: 600px}</style>';
         include_once $pth['folder']['plugins'] . 'jquery/jquery.inc.php';
         include_jQuery();
         include_jQueryPlugin('jstree', "{$this->pluginFolder}jstree/jstree.min.js");
`````

Aber die Hilfe zu `$cf['editor']['height']` sagt:

> Integer oder JavaScript-Ausdruck, der einen Integer liefert für die Editorhöhe in Pixel.

JS-Ausdrücke kann ich aber so nicht nutzen. Bliebe die Möglichkeit auf einen Integer-Wert zu prüfen, und bei Erfolg diesen zu verwenden, und sonst beim Stylesheet-Wert zu bleiben. Oder eine eigene Konfigurationsoption (finde ich nicht so prickelnd).

Im Zweifel würde ich es für Version 3.7 einfach so lassen wie es ist. Was meinst du, @frase-git?

**Comment by frase-git** [comment]
Dass man einen JS-Ausdruck für die Editor-Höhe schreiben könnte, war mir bisher gar nicht bewusst.
Ich kann mir auch gar nicht vorstellen, warum und bei welcher Gelegenheit man das tun sollte.
Da sollte letztendlich vielleicht immer nur eine Ganzzahl erlaubt sein.

Die Höhe des Pagemanagers jetzt extra hart-kodiert einzuschleusen, bringt auch nicht viel.

Aber egal, so richtig wichtig ist mir das alles nicht. Es müsste halt noch ein zusätzliches Label geben, wie z. B. „Anregung/vielleicht“. ;-)

Deshalb: Ja, für Version 3.7 einfach so lassen wie es ist.

**Comment by cmb69** [comment]
> Dass man einen JS-Ausdruck für die Editor-Höhe schreiben könnte, war mir bisher gar nicht bewusst.
> Ich kann mir auch gar nicht vorstellen, warum und bei welcher Gelegenheit man das tun sollte.

Die Idee war, dass man den verfügbaren Platz eben genau berechnen kann, statt einfach einen festen Wert zu vergeben. Ob das aber wirklich von allen Editoren unterstützt wird, weiß ich gar nicht.

> Deshalb: Ja, für Version 3.7 einfach so lassen wie es ist.

Okay, dann stelle ich das gerne zurück.
